### DISCUSSION TITLE
Clerk Auth Documentation

### DISCUSSION THREAD

**Discussion by natiwho** [withastro/starlight#2499] [Q&A] [answered]
### What version of `starlight` are you using?

0.28.3

### What is your idea?

Could you please share documentation on how to integrate an authentication solution like Clerk for private documentation easily? 

If not yet possible can this be a feature for the next release?

### Why is this feature necessary?

Security for private docs

### Do you have examples of this feature in other projects?

It's as straightforward as securing some documents for private use.

Imagine a developer's documentation organized by folder structure, where some folders contain private projects and others hold public ones.

You can see an example at:
[docs.natnael.dev](https://docs.natnael.dev/)

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment, marked as answer]
Thanks for the feedback :raised_hands: 

Linking to https://github.com/withastro/starlight/discussions/1348 which is a discussion on the entire topic of authentication. It contains a link to [TheOtterlord/starlight-auth](https://github.com/TheOtterlord/starlight-auth) which is a Starlight template that supports e-mail signin, traditional username/password auth, and 80+ OAuth providers using the [Auth.js](https://authjs.dev/) library.

A few extra links:

- The [authentication](https://docs.astro.build/en/guides/authentication/) guide in the Astro Docs as Starlight is built on top of Astro so most of the concepts are the same.
- An [example](https://github.com/HiDeoo/starlight-better-auth-example) of a Starlight website with authentication using the [Better Auth](https://www.better-auth.com/) library that I recently created where users can create an account, sign in, and sign out. Some pages are protected and require the user to be authenticated to access them or even be visible in the documentation sidebar.
